### ISSUE TITLE
ssh_setup.sh has system accounts hard-coded as being below UID=500

### ISSUE THREAD

**Issue by jtoddowen** [hpcng/warewulf#552]
In file `overlays/host/etc/profile.d/ssh_setup.sh` and `.csh`, UIDs below 500 are considered system accounts.

In lieu of hard-coding this, one could use something like the following to determine the installation specific UID numeric value:
`` stringZ=`cat /etc/login.defs | grep UID_MIN`
expr match "$stringZ" '.*\([0-9]*\)' ``

This should pull out the numeric value, whether three, four or five digits, and use it instead.  Newer distributions use 1000 as a starting point for real users.  I start at 2001 for Faculty and Staff users and 3001 for students.

**Comment by anderbubble** [comment]
Agreed by community discussion. `ssh_setup.sh` could probably use and audit overall.

@griznog is going to take a look.

**Comment by griznog** [comment]
@anderbubble it didn't take long to reach some indecision.

1. This is currently in the host overlay, which assumes the master is the login node. That is only true for trivial cluster setups though. For sites that provision login nodes via WW this should probably be in a node runtime overlay. But that breaks the simple default setup.  This is a good example of where splitting things across multiple overlays is useful, if there were a `login-node` overlay or a `profile.d` overlay and the host was just a regular node in the config then this and related items could go there and the overlay could be assigned as needed, whether to the host or to a provisioned login node. 

2. I was hoping to use `tags` to set `min_uid` and possibly some other values to use in the script so that it'd be easy for admins to set values then fall back to pulling from `/etc/login.defs` if no value was explicitly set. But it's not apparent to me how/where I could get tags into the host config to use in templates. See previous point as this would be trivial if the host was Just Another Node(tm) in `nodes.conf`. 

3. There is a `.csh` version of the script, does anyone still use `[t]csh` as a login shell? I'd have to somehow reawaken long dormant neurons to keep the functionality of the `.sh` and `.csh` versions comparible :). 

4. Looking at the  existing script, an improved one would be a do-over sparking a general question about the license. If replacing this with a completely new version, does the license part need to be preserved? It's never been clear to me how much change is required to make the new thing not a derivative of the old thing w.r.t. the license.

**Comment by anderbubble** [comment]
I think a new one would just be licensed under the general project license, and wouldn't have to inherit whatever is in the current file.

Agreed on the value of breaking up the host overlay and supporting this on other nodes than the Warewulf host / server.

I _do_ think we should retain the `.csh` functionality. At least on Rocky this matches practice, given a glance at `/etc/profile.d/`.

Using tags may be nice, but just pulling from `login.defs` would probably be sufficient for now.
